.github/workflows/tests.yml

```diff
@@ -62,10 +62,18 @@ jobs:
       - name: Run unit tests
         run: make unit-tests
 
-      # disabled as 'pinging codecov' hangs
-      # see https://github.com/codecov/feedback/issues/354
-      # - name: Send coverage to codecov.io
-      #   run: bash <(curl -s https://codecov.io/bash)
+  codecov:
+    name: Run tests and collect coverage
+    runs-on: ubuntu-latest
+    steps:
+     - uses: actions/checkout@v2
+       with:
+         fetch-depth: 2
+     - name: Upload results to Codecov
+       uses: codecov/codecov-action@v4
+       with:
+         files: ./coverage.txt
```

Comment:
I think, now the `make unit-tests` is missing (which is generating the `coverage.txt`)
and probably best to run it only once (as @ezr-ondrej suggested) like it was before commenting it out

so just replacing the comment with
```
     - name: Upload results to Codecov
       uses: codecov/codecov-action@v4
       with:
         files: ./coverage.txt
         token: ${{ secrets.CODECOV_TOKEN }}
```
might work

Reply:
looks like it is working 
```info - 2024-08-19 19:28:51,723 -- Found 1 coverage files to report
info - 2024-08-19 19:28:51,723 -- > /home/runner/work/image-builder/image-builder/coverage.txt
info - 2024-08-19 19:28:53,258 -- Your upload is now processing. When finished, results will be available at: https://app.codecov.io/github/osbuild/image-builder/commit/4245f229d5c9be4e41c5770e2d10bb4059d7ea7a
info - 2024-08-19 19:28:53,466 -- Process Upload complete```